src/modules/sway/mode.rs

```diff
@@ -0,0 +1,94 @@
+use crate::config::{CommonConfig, TruncateMode};
+use crate::modules::{Module, ModuleInfo, ModuleParts, ModuleUpdateEvent, WidgetContext};
+use crate::{await_sync, glib_recv, module_impl};
+use color_eyre::{Report, Result};
+use gtk::prelude::*;
+use gtk::Label;
+use serde::Deserialize;
+use swayipc_async::{Event, EventType, ModeEvent};
+use tokio::sync::mpsc;
+use tracing::{info, trace};
+
+#[derive(Debug, Deserialize, Clone)]
+#[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
+pub struct SwayModeModule {
+    // -- Common --
+    /// See [truncate options](module-level-options#truncate-mode).
+    ///
+    /// **Default**: `null`
+    pub truncate: Option<TruncateMode>,
+
+    /// See [common options](module-level-options#common-options).
+    #[serde(flatten)]
+    pub common: Option<CommonConfig>,
+}
+
+impl Module<Label> for SwayModeModule {
+    type SendMessage = ModeEvent;
+    type ReceiveMessage = ();
+
+    module_impl!("sway_mode");
+
+    fn spawn_controller(
+        &self,
+        _info: &ModuleInfo,
+        context: &WidgetContext<Self::SendMessage, Self::ReceiveMessage>,
+        _rx: mpsc::Receiver<Self::ReceiveMessage>,
+    ) -> Result<()> {
+        info!("Sway Mode module started");
+        let tx = context.tx.clone();
+
+        await_sync(async move {
+            let client = context.ironbar.clients.borrow_mut().sway()?;
+            client
+                .add_listener(
+                    EventType::Mode,
+                    Box::new(move |event| {
+                        trace!("event: {:?}", event);
+                        let Event::Mode(mode) = event else {
+                            unreachable!()
+                        };
+                        let _ = await_sync(tx.send(ModuleUpdateEvent::Update(mode.clone())));
```

Comment:
This can be simplified using `crate::try_send`:
```suggestion
                        try_send!(tx, ModuleUpdateEvent::Update(mode.clone()))
```

(I think I counted those brackets right...)